kite-data/kite-data-core/src/main/java/org/kitesdk/data/Flushable.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2013 Cloudera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kitesdk.data;
+
+/**
+ * A writer that can guarantee data is persisted to disk.
+ * <p>
+ * {@link #flush} a stronger guarantee than {@link java.io.Flushable#flush}.
+ * When it returns, data written is persisted to disk, not just an underlying
+ * buffer.
+ */
+@edu.umd.cs.findbugs.annotations.SuppressWarnings(
+    value="NM_SAME_SIMPLE_NAME_AS_INTERFACE",
+    justification="Intended to be a stricter version of the parent interface")
+public interface Flushable extends java.io.Flushable {
+  /**
+   * Ensure that data has been written to disk.
+   * <p>
+   * This method has a stronger guarantee than {@link java.io.Flushable#flush}.
+   * When this method returns, data written is persisted to disk, not just an
+   * underlying buffer.
+   */
+  @Override
+  void flush();
```

Comment:
The pattern in all of our writers is to call flush before calling sync because hsync doesn't necessarily hflush; sync is effectively hflush+hsync. At the very least, we need to be clear that a call to sync is a call to flush so we don't have callers flushing twice. From there, I started to question whether it was worth having two methods for this in the API.

A single method that provides durability is easier to use in terms of both 1) not having to read docs about why there are two and 2) not making a mistake and calling flush when you mean to sync. And, If sync always calls hflush, then how valuable is flush on its own? Execution returns earlier at the cost of not ensuring data is actually on disk, so anyone writing, say, a Flume Sink will won't be able to take advantage of flush because they must call hsync before committing the current transaction.

I think what little utility is gained by exposing hflush alone is outweighed by the confusion of two methods, one of which seems reliable but is really not. Is that reasonable?